python/ray/tune/tests/test_basic_variant.py

```diff
@@ -0,0 +1,130 @@
+import copy
+import unittest
+
+from ray import tune
+from ray.tune.experiment import Experiment
+from ray.tune.search.basic_variant import BasicVariantGenerator
+
+
+class BasicVariantGeneratorTest(unittest.TestCase):
+    """
+    Test BasicVariantGenerator.
+    """
+
+    def testSanityCheck(self):
+        gen = BasicVariantGenerator(
+            points_to_evaluate=[
+                {"a": 4, "b": {"d": 6}},
+                {"a": 3},
+                {"c": 8},
+            ],
+            max_concurrent=1,  # 1 trial at a time.
+        )
+        gen.add_configurations(Experiment(
+            name="test",
+            run=lambda config: config["a"] + config["b"],
+            config={
+                "a": tune.grid_search([3, 4]),
+                "b": {
+                    "d": tune.choice([6, 8]),
+                },
+                "c": tune.randint(1, 10),
+            }
+        ))
+
+        trial_idx = 0
+        trial = gen.next_trial()
+        while trial:
+            # Let generator know that trial is finished.
+            gen.on_trial_complete(trial.trial_id, result=0)
+
+            # Check a.
+            self.assertTrue(trial.config["a"] in [3, 4])
+
+            # Check b/d.
+            self.assertTrue(trial.config["b"]["d"] in [6, 8])
+
+            # Check c.
+            self.assertGreaterEqual(trial.config["c"], 1)
+            self.assertLessEqual(trial.config["c"], 10)
+
+            trial = gen.next_trial()
+            trial_idx += 1
+
+        # 4 Trials in total.
+        # Top 2 trials form points_to_evaluate.
+        # The third points_to_evaluate point will create 2 trials, where
+        # 'a' will sample both values.
+        self.assertEqual(trial_idx, 4)
+
+    def testParamSpaceOverride(self):
+        opt1 = {}
```

Comment:
These should all be some dummy object `Test()`. The `{} in [{}, {}]` will always be true in the asserts later.

Reply:
ok